File: JSFML-PONG/src/pong/Rectangle.java
package pong;

import org.jsfml.graphics.*;
import org.jsfml.system.*;

public class Rectangle {

	protected RectangleShape Object = new RectangleShape();

	public// Konstruktor
	Rectangle(int WIDTH, int HEIGHT, Color COLOR) {
		Object.setSize(new Vector2f(WIDTH, HEIGHT));
		Object.setFillColor(COLOR);

	};

	// set position
	void setPosition(Vector2f POSITION) {
		Object.setPosition(POSITION);
	};

	void setColor(Color color) {
		Object.setFillColor(color);
	};

	// sprawdza czy dwa obiekty sie dotykaja
	boolean touches(RectangleShape RECTANGLE) {
		if (Object.getGlobalBounds().intersection(RECTANGLE.getGlobalBounds()) != null)
			return true;
		return false;
	};

	Vector2f getPosition() {
		return Object.getPosition();
	};

	Vector2f getSize() {
		return Object.getSize();
	};

	RectangleShape getObject() {
		return Object;
	};

}

File: SFML-PONG/src/Rectangle.cpp
#include "../include/Rectangle.h"

Rectangle::Rectangle(int WIDTH, int HEIGHT, sf::Color COLOR)
{
    Object.setSize(sf::Vector2f(WIDTH, HEIGHT));
    Object.setFillColor(COLOR);
}

sf::RectangleShape Rectangle::getObject()
{
    return Object;
};

void Rectangle::setColor( sf::Color color)
{
    Object.setFillColor(color);
};


void Rectangle::setPosition(sf::Vector2f POSITION)
{
    Object.setPosition(POSITION);
}


bool Rectangle::touches(sf::RectangleShape RECTANGLE)
{
    if(Object.getGlobalBounds().intersects(RECTANGLE.getGlobalBounds()))
        return true;
    return false;
}

sf::Vector2f Rectangle::getPosition()
{
    return Object.getPosition();
}

sf::Vector2f Rectangle::getSize()
{
    return Object.getSize();
}

Rectangle::~Rectangle()
{

}
